.github/workflows/multi-approvers.js

```diff
@@ -2,49 +2,120 @@ const APPROVED = 'APPROVED';
+        if (reviewStateByLogin[reviewerLogin] !== APPROVED) {
+          reviewStateByLogin[reviewerLogin] = r.state;
+          return;
+        }
+
+        // Do not update approved state for a comment.
+        if (reviewStateByLogin[reviewerLogin] === APPROVED && r.state !== COMMENTED) {
+          reviewStateByLogin[reviewerLogin] = r.state;
+        }
+      })
+
+    return Object.values(reviewStateByLogin).filter((s) => s === APPROVED).length;
+  }
+}
 
-      // Always update state if not approved.
-      if (reviewStateByLogin[reviewerLogin] !== APPROVED) {
-        reviewStateByLogin[reviewerLogin] = r.state;
-        return;
-      }
+/** Members backed by a JSON file. */
+class JsonMembers extends AbstractMembers {
+  members;
 
-      // Do not update approved state for a comment.
-      if (reviewStateByLogin[reviewerLogin] === APPROVED && r.state !== COMMENTED) {
-        reviewStateByLogin[reviewerLogin] = r.state;
-      }
-    })
+  constructor(membersPath) {
+    super();
+    this.members = require(membersPath).reduce((acc, v) => acc.set(v.login, v), new Map());
+  }
 
-  return Object.values(reviewStateByLogin).filter((s) => s === APPROVED).length;
+  async contains(login) {
+    return this.members.has(login);
+  }
+}
+
+/** Members backed by a GitHub team. */
+class TeamMembers extends AbstractMembers {
+  static ALLOWED_ROLES = ["maintainer", "member"];
+  static ACTIVE = "active";
+
+  org;
+  teamSlug;
+  github;
+
+  constructor(org, teamSlug, github) {
+    super();
+    this.org = org;
+    this.teamSlug = teamSlug;
+    this.github = github;
+  }
+
+  async contains(login) {
+    try {
+      const response = await this.github.rest.teams.getMembershipForUserInOrg({
+        org: this.org,
+        team_slug: this.teamSlug,
+        username: login,
+      });
+      return TeamMembers.ALLOWED_ROLES.indexOf(response.data.role) >= 0 &&
+          response.data.state === TeamMembers.ACTIVE;
+    } catch (error) {
+      if (error.status === 404) {
+        // We can get here for two reasons:
+        // 1) The user is not a member
+        // 2) The team does not exist
+        // Either way, it's safe to return false here.
+        return false;
+      }
+      throw error;
+    }
+  }
 }
 
 /** Checks that approval requirements are satisfied. */
-async function onPullRequest({orgMembersPath, prNumber, repoName, repoOwner, github, core}) {
-  const members = require(orgMembersPath).reduce((acc, v) => acc.set(v.login, v), new Map());
+async function onPullRequest({orgTeam, membersPath, prNumber, repoName, repoOwner, github, core}) {
+  const members = (function() {
+    if (orgTeam) {
+      if (orgTeam.indexOf('/') <= 0) {
+        throw new Error(`Malformed team [${orgTeam}]. Team must be in the format \${org}/\${team-slug}.`);
+      }
+      const teamOrg = orgTeam.split('/')[0];
+      const teamSlug = orgTeam.split('/')[1];
+      return new TeamMembers(teamOrg, teamSlug, github);
```

Comment:
Double split is inefficient. This will also potentially be `undefined` if the `orgTeam` doesn't contain a slash.

Reply:
No, this is a bug that will cause a runtime panic and/or unexpected behavior. It needs to be fixed prior to merging. We do not merge known-bugs into HEAD.